### PR TITLE
Add dimenstion for arm packaging

### PR DESCRIPTION
**Pull request by XilaiZhang**
context:  https://github.com/flutter/flutter/issues/122769

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -4964,6 +4964,8 @@ targets:
       task_name: flutter_packaging
       tags: >
         ["framework", "hostonly", "shard", "mac"]
+    dimensions:
+      cpu: "arm64"
```
Comment (christopherfujino) [inline .ci.yaml:4968]: I'll defer final approval. This file is confusing however, I see arm64 being specified under `cpu: arm64` https://github.com/flutter/flutter/blob/master/.ci.yaml#L90 and `properties.tags: ["arm64"]` https://github.com/flutter/flutter/blob/master/.ci.yaml#L3331, and then here under `dimensions.cpu: arm64`.

Comment (XilaiZhang) [reply .ci.yaml:4968]: Umm that's true, if it is specified in the general properties it should have been inherited

Comment (godofredoc) [reply .ci.yaml:4968]: To keep the dart internal builds simple they do not implement the logic post-process the ci.yaml file. They only see the specific target information.

**Review by godofredoc — APPROVED** [review]

File: .ci.yaml (hunk 1 above)
Comment (godofredoc) [reply .ci.yaml:4968]: By the way we have an open bug to simplify properties and dimensions across flutter-infra, ci.yaml and cocoon. Turns out we have different implementations on those 3 places.

Comment (XilaiZhang) [reply .ci.yaml:4968]: Thanks for explaining! Didn't know dart internal uses a different parser for yaml

**Pull request merged**